apps/etterlatte-behandling/src/main/kotlin/grunnlagsendring/doedshendelse/kontrollpunkt/DoedshendelseKontrollpunktService.kt

```diff
@@ -34,23 +39,156 @@ class DoedshendelseKontrollpunktService(
+        avdoed: PersonDTO,
+    ): DoedshendelseKontrollpunkt.EpsHarVaertSkiltSiste5MedUkjentGiftemaalLengde? {
+        if (eps.sivilstand != null) {
+            val skilt = eps.sivilstand!!.map { it.verdi }.filter { it.sivilstatus == Sivilstatus.SKILT }
+            val skiltMedAvdoed = skilt.filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }
+            val naa = LocalDate.now()
+            val gift = eps.sivilstand!!.map { it.verdi }.filter { it.sivilstatus == Sivilstatus.GIFT }
+            val giftMedAvdoed = gift.filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }
+            if (skiltMedAvdoed.isNotEmpty() && giftMedAvdoed.isNotEmpty()) {
+                val skiltsivilstand = skiltMedAvdoed.first()
+                val antallSkilteAar = safeYearsBetween(skiltsivilstand.gyldigFraOgMed, naa).absoluteValue
+                val giftSivilstand = giftMedAvdoed.first()
+                return if (antallSkilteAar <= 5 && giftSivilstand.gyldigFraOgMed == null) {
+                    DoedshendelseKontrollpunkt.EpsHarVaertSkiltSiste5MedUkjentGiftemaalLengde
+                } else {
+                    null
+                }
+            } else {
+                return null
+            }
+        } else {
+            return null
+        }
+    }
+
+    private fun kontrollerSkiltSistefemAarEllerGiftI15(
+        eps: PersonDTO,
+        avdoed: PersonDTO,
+    ): DoedshendelseKontrollpunkt.EpsHarVaertSkiltSiste5EllerGiftI15? {
+        if (eps.sivilstand != null) {
+            val skilt = eps.sivilstand!!.map { it.verdi }.filter { it.sivilstatus == Sivilstatus.SKILT }
+            val skiltMedAvdoed = skilt.filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }
+            val gift = eps.sivilstand!!.map { it.verdi }.filter { it.sivilstatus == Sivilstatus.GIFT }
+            val giftMedAvdoed = gift.filter { it.relatertVedSiviltilstand?.value == avdoed.foedselsnummer.verdi.value }
+            val naa = LocalDate.now()
+            if (skiltMedAvdoed.isNotEmpty() && giftMedAvdoed.isNotEmpty()) {
+                val skiltsivilstand = skiltMedAvdoed.first()
+                val antallSkilteAar = safeYearsBetween(naa, skiltsivilstand.gyldigFraOgMed).absoluteValue
+                val giftSivilstand = giftMedAvdoed.first()
+                val antallGifteAar = safeYearsBetween(giftSivilstand.gyldigFraOgMed, naa).absoluteValue
+                return if (antallSkilteAar <= 5 && antallGifteAar >= 15) {
+                    DoedshendelseKontrollpunkt.EpsHarVaertSkiltSiste5EllerGiftI15
+                } else {
+                    null
+                }
+            } else {
+                return null
+            }
+        } else {
+            return null
+        }
+    }
+
+    private fun kontrollerEpsHarFylt67Aar(eps: PersonDTO): DoedshendelseKontrollpunkt.EpsKanHaAlderspensjon? {
+        return if (eps.foedselsdato != null) {
```

Comment:
Skulle vi gjort noe sånt her kanskje: `foedselsdato ?: LocalDate.of(foedselsaar, 1, 1)`?

Reply:
hmm